modules/cloudevent-recorder/variables.tf

```diff
@@ -52,5 +52,6 @@ variable "types" {
     schema                = string
     alert_threshold       = optional(number, 50000)
     notification_channels = optional(list(string), [])
+    partition_field       = optional(string)
```

Comment:
```suggestion
    partition_field       = optional(string, "")
```

Reply:
I don't think we want this - we want to default to null if the field isn't set so we keep compatibility to the current behavior.

Setting "" may cause the GCP API to treat this as a set value and return an error.

If you want I can make this `optional(string, null)` to be explicit?